dotCMS/src/main/java/com/dotcms/cluster/business/ServerAPIImpl.java

```diff
@@ -35,52 +44,111 @@ public void saveServer(Server server) throws DotDataException {
+                newServer.setEsHttpPort(Integer.parseInt(port));
+    
+               updateServer(newServer);
+    
+                try {
+                    writeHeartBeatToDisk(newServer.getServerId());
+                } catch (IOException e) {
+                    Logger.error(ClusterFactory.class, "Could not write Server ID to file system" , e);
+                }
+       
+            });
 
-	public String readServerId() {
-
-		String realPath = ConfigUtils.getDynamicContentPath() + java.io.File.separator + "server_id.dat";
-
-		Logger.debug(ServerAPIImpl.class, "Reading " + realPath);
-
-		BufferedReader br = null;
-		String serverId = null;
-		try {
-			br = new BufferedReader(new FileReader(new File(realPath)));
-			serverId = br.readLine();
-		} catch (FileNotFoundException e) {
-			Logger.debug(ServerAPIImpl.class, "Server ID not found (" + realPath + " file doesn't exists)");
-		} catch (IOException e) {
-			Logger.error(ServerAPIImpl.class, "Could not read Server ID from File", e);
-		} finally {
-			try {
-				if(br!=null)
-					br.close();
-			} catch (IOException e) {
-				Logger.error(ServerAPIImpl.class, "Could not close BufferedReader for Server File", e);
-			}
-		}
-
-		Logger.debug(ServerAPIImpl.class, "ServerID: " + serverId);
-
-        return serverId;
-
-	}
+        }
+        
 
-	public  void writeServerIdToDisk(String serverId) throws IOException {
+        return serverFactory.getServer(serverId);
+    }
 
+	private File serverIdFile(){
+	    
         String dynamicContentPath = ConfigUtils.getDynamicContentPath();
 
+        
         String realPath;
         if ( dynamicContentPath.endsWith( File.separator ) ) {
             realPath = ConfigUtils.getDynamicContentPath() + "server_id.dat";
         } else {
             realPath = ConfigUtils.getDynamicContentPath() + File.separator + "server_id.dat";
         }
-        File serverFile = new File(realPath);
+        Logger.debug(ServerAPIImpl.class, "Server Id " + realPath);
 
-		if(serverFile.exists())
-			serverFile.delete();
+        return new File(realPath);
 
+	}
+	
+	
+	
+	
+	@Override
+	public String readServerId()  {
+	    // once set this should never change
+	    if(SERVER_ID==null){
+    	    try{
+        	    File serverFile = serverIdFile();
+        
+        	    if(!serverFile.exists()){
+        	        writeServerIdToDisk(UUIDUtil.uuid());
+        	    }
+        
+        		BufferedReader br =  new BufferedReader(new FileReader(serverFile));
+        		SERVER_ID = br.readLine();
+        		br.close();
+        
+        		Logger.debug(ServerAPIImpl.class, "ServerID: " + SERVER_ID);
+        
+    
+    	    }
+    	    catch(IOException ioe){
+    	        throw new DotStateException("Unable to read server id at " + serverIdFile() + " please make sure that the directory exists and is readable and writeable. If problems persist, try deleting the file.  The system will recreate a new one on startup", ioe);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 265 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)